src/citrine/builders/auto_configure.py

```diff
@@ -0,0 +1,183 @@
+        {'properties': [...], 'conditions': [...], 'parameters': []}
+
+    """
+    if mode not in {"FORMULATION", "SIMPLE"}:
+        msg = "Called with mode: {}.  Expected 'SIMPLE' or 'FORMULATION'.".format(mode)
+        raise ValueError(msg)
+
+    link = _make_link_by_uid(material)
+    history = project.material_runs.get_history(scope=link.scope, id=link.id)
+
+    if full_history:
+        search_history = flatten(history)
+    else:
+        # Limit the search to contain the terminal material/process/measurement
+        search_history = [history.spec.template, history.process.template]
+        search_history.extend([msr.template for msr in history.measurements])
+
+    keys = {'properties': [], 'conditions': [], 'parameters': []}
+    for obj in search_history:
+        parent_name = f'{obj.name}~' if mode == 'SIMPLE' else ''
+        # Search all attribute template types
+        if isinstance(obj, HasPropertyTemplates):
+            for property in obj.properties:
+                keys['properties'].append(parent_name + property[0].name)
+        if isinstance(obj, HasConditionTemplates):
+            for condition in obj.conditions:
+                keys['conditions'].append(parent_name + condition[0].name)
+        if isinstance(obj, HasParameterTemplates):
+            for parameter in obj.parameters:
+                keys['parameters'].append(parent_name + parameter[0].name)
+    return keys
+
+
+def auto_configure_material(
+        *,
+        project: Project,
+        material: Union[MaterialRun, LinkByUID, str, UUID],
+        score: Score,
+        mode: str = 'SIMPLE',
+        label: str = '',
+        print_status_info: bool = False,
+) -> DesignExecution:
+    """[ALPHA] Description.
+
+    Parameters
+    ----------
+    project: Project
+        Project to use when accessing the Citrine Platform
+    material: Union[MaterialRun, LinkByUID, str, UUID]
+        Sample material to extract descriptor keys
+    score: Score
+        Scoring function used to rank candidates during design execution.
+        Must contain objectives/constraints with matching descriptor keys
+        to those found within the provided material history.
+    mode: str
+        Option in {'SIMPLE', 'FORMULATION'} that controls formatting descriptor keys
+        Default: 'SIMPLE'
+    label: str
+        Naming label to affix to auto-configured assets on the Citrine Platform
+        Default: ''
+    print_status_info: bool
+        Whether to print the status of predictor, design space, and design workflow validation
+        Default: False
+
+    Returns
+    -------
+    DesignExecution
+        Triggered design execution given the auto-configured workflow and provided `score`
+
+    """
+    if mode not in {"FORMULATION", "SIMPLE"}:
+        msg = "Called with mode: {}.  Expected 'SIMPLE' or 'FORMULATION'.".format(mode)
+        raise ValueError(msg)
+
+    suffix = ' - {}'.format(label) if label else ''
```

Comment:
We might want to use the label as a prefix, otherwise multiple invocations willr esult in a bunch of "Default {Asset Name}" in the lists.